### DISCUSSION TITLE
Is there a way to give template or workflow a time to run ? (shutdown workflow in a given time if still running)

### DISCUSSION THREAD

**Discussion by mozarik** [argoproj/argo-workflows#11940] [Q&A] [answered]
Do you guys know how to do this ? let say i have a workflow A. In that workflow i want to set Workflow A should run under 30 minutes. If its still running exceed 30 minutes it will shutdown/terminate (or retry) the workflow.

If i can do that. Can i also do that to a specific steps or dag ?

**Comment by agilgur5** [comment, marked as answer]
You can use [`activeDeadlineSeconds`](https://argoproj.github.io/argo-workflows/walk-through/timeouts/) to set a timeout. It can be set at both the [Workflow-level](https://argoproj.github.io/argo-workflows/fields/#workflowspec) and the [template-level](https://argoproj.github.io/argo-workflows/fields/#template)

**Reply by mozarik** [reply]
Thank you. this is what I looking for
